src/main/kotlin/com/devh/vitstore/service/product/ProductService.kt

```diff
@@ -0,0 +1,31 @@
+package com.devh.vitstore.service.product
+
+import com.devh.vitstore.model.product.ProductDao
+import com.devh.vitstore.model.product.ProductDto
+import com.devh.vitstore.repository.ProductRepository
+import javassist.NotFoundException
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+import javax.transaction.Transactional
+
+@Service
+@Transactional
+class ProductService {
+    @Autowired
```

Comment:
sao vẫn còn e?

Reply:
@huorlk-0450 đây là service mà a. service JwtUserDetailsService a cũng viết thế mà

Reply:
nó dùng trong config chưa biết cách fix để dùng kiểu constructor

Reply:
https://www.baeldung.com/constructor-injection-in-spring#pros-and-cons